crates/bitwarden-exporters/src/cxp/export.rs

```diff
@@ -0,0 +1,454 @@
+            // TODO(PM-15450): Add support for credit cards.
+            CipherType::Card(_) => vec![],
+            // TODO(PM-15451): Add support for identities.
+            CipherType::Identity(_) => vec![],
+            // Secure Notes only contains a note field which is handled by `TryFrom<Cipher> for
+            // Item`.
+            CipherType::SecureNote(_) => vec![],
+            // TODO(PM-15448): Add support for SSH Keys.
+            CipherType::SshKey(_) => vec![],
+        }
+    }
+}
+
+/// Convert a `Login` struct into the appropriate `Credential`s.
+impl From<Login> for Vec<Credential> {
+    fn from(login: Login) -> Self {
+        let mut credentials = vec![];
+
+        if login.username.is_some() || login.password.is_some() || !login.login_uris.is_empty() {
+            credentials.push(Credential::BasicAuth(Box::new(login.clone().into())));
+        }
+
+        if let Some(totp) = login.totp.and_then(|t| t.parse::<Totp>().ok()) {
+            credentials.push(Credential::Totp(Box::new(convert_totp(totp))));
+        }
+
+        if let Some(fido2_credentials) = login.fido2_credentials {
+            credentials.extend(
+                fido2_credentials
+                    .into_iter()
+                    .filter_map(|fido2_credential| fido2_credential.try_into().ok())
+                    .map(|c| Credential::Passkey(Box::new(c))),
+            );
+        }
+
+        credentials
+    }
+}
+
+/// Convert a `Totp` struct into a `TotpCredential` struct
+fn convert_totp(totp: Totp) -> TotpCredential {
+    // TODO(PM-15389): Properly set username/issuer.
+    TotpCredential {
+        secret: totp.secret.into(),
+        period: totp.period as u8,
+        digits: totp.digits as u8,
+        username: "".to_string(),
+        algorithm: match totp.algorithm {
+            TotpAlgorithm::Sha1 => OTPHashAlgorithm::Sha1,
+            TotpAlgorithm::Sha256 => OTPHashAlgorithm::Sha256,
+            TotpAlgorithm::Sha512 => OTPHashAlgorithm::Sha512,
+            TotpAlgorithm::Steam => OTPHashAlgorithm::Unknown("steam".to_string()),
+        },
+        issuer: None,
+    }
+}
+
+impl From<Login> for BasicAuthCredential {
+    fn from(login: Login) -> Self {
+        BasicAuthCredential {
+            urls: login
+                .login_uris
+                .into_iter()
+                .flat_map(|uri| uri.uri)
+                .collect(),
+            username: login.username.map(|value| EditableField {
+                id: random_id(),
+                field_type: FieldType::String,
+                value,
+                label: None,
+            }),
+            password: login.password.map(|value| EditableField {
+                id: random_id(),
+                field_type: FieldType::ConcealedString,
+                value,
+                label: None,
+            }),
+        }
+    }
+}
+
+/// Generate a 32 byte random ID
+///
+/// TODO: This should be removed shortly.
+fn random_id() -> B64Url {
```

Comment:
Question: What do you mean this should be removed shortly? Will the `id` field in `EditableField` be optional or do we plan to have a better way to generate these ids?

Reply:
It will become optional.